jobs/garbage-collection.Jenkinsfile

```diff
@@ -0,0 +1,100 @@
+                memory: "1024Mi",
+                serviceAccount: "jenkins") {
+            timeout(time: 30, unit: 'MINUTES') {
+                try {
+                    // Write YAML data to a new file in cosaPod
+                    def new_gc_policy_path = './gc-policy.yaml'
+                    writeYaml file: new_gc_policy_path, data: gc_policy_data
+
+                    currentBuild.description = "${build_description} Running"
+
+                    def credentials = [file(variable: 'GCP_KOLA_TESTS_CONFIG', credentialsId: 'gcp-image-upload-config')]
+
+                    withCredentials(credentials) {
+                        // pick up the GCP project to use from the config
+                        def gcp_project = shwrapCapture("jq -r .project_id \${GCP_KOLA_TESTS_CONFIG}")
+                        def acl = pipecfg.s3.acl ?: 'public-read'
+                        def dry_run = params.DRY_RUN ? "--dry-run" : ""
+                        
+                        stage('Init') {
+                            def branch = params.STREAM
+                            sh "cosa init --force --branch ${branch} https://github.com/${repo}"
+                        }
+                        
+                        // buildfetch for the specified stream to get the respective  builds.json
+                        stage('BuildFetch') {
+                            pipeutils.shwrapWithAWSBuildUploadCredentials("""
+                            cosa buildfetch --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG} \
+                                --stream ${params.STREAM} --url=s3://${s3_stream_dir}/builds
+                            """)
+                        }
+                        
+                        // Run cloud-prune to delete the resources specified in the gc-policy.yaml
+                        stage('Garbage Collection') {
+                            pipeutils.shwrapWithAWSBuildUploadCredentials("""
+                            cosa cloud-prune --policy ${new_gc_policy_path} \
+                            --stream ${params.STREAM} ${dry_run} \
+                            --gcp-project=${gcp_project} \
+                            --gcp-json-key=\${GCP_KOLA_TESTS_CONFIG} \
+                            --acl=${acl} \
+                            --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG}
+                            """)
+                        }
+                        
+                        // Upload the updated builds.json back to s3 bucket
+                        stage('Upload Builds JSON') {
+                            pipeutils.shwrapWithAWSBuildUploadCredentials("""
+                            cosa cloud-prune --policy ${new_gc_policy_path} \
+                            --stream ${params.STREAM} \
+                            --upload-builds-json ${dry_run} \
+                            --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG}
+                            """)
+                        }
```

Comment:
This needs to be done under the same lock taken by `bump_builds_json()`. (Let's rename that lock there to just e.g. `builds-json-${stream}`.)